src/events.py

```
@@ -18,8 +19,57 @@ def apply_event(self, player):
             #set the player item to the current value plus the added or subtracted value.
 
             if key == "damage" or key == "huge_damage":
-                setattr(player, 'health', getattr(player, 'health', 0) + value)
-                display_warning_message(f"{self.description}: {value}")
+                if "weapon" in player.inventory.items.keys() and player.inventory.items["weapon"] > 0:
```

Ruff (0.8.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)